org.eclipse.lyo.oslc4j.codegenerator/src/org/eclipse/lyo/oslc4j/codegenerator/files/generateServiceProviderCatalogSingleton.mtl

```diff
@@ -181,7 +181,7 @@ public class [javaClassNameForSingleton(aServiceProviderCatalog) /]
 
     private static String [aServiceProvider.javaName(false)/]Identifier([aServiceProvider.instanceMethodSignature() /])
```

Comment:
@berezovskyi  This explains the reason for the seemingly unnecessary method `serviceProviderIdentifier(final String serviceProviderId)` that simply returns the paramter serviceProviderId.

For the very unlikley SPs that have a composite id (such as "/x/{x}/y/{y}" I needed to produce a unique key. 
Most cases don't require such composite IDs, but there was such a need in the past, and now we have it.